### PR TITLE
Update kustomize to use overlays

### PR DESCRIPTION
**Pull request by arielireni**
Replaced adding/removing resources in `make gen-manifests` with the use of kustomize base and overlays for each cluster type.

Signed-off-by: arielireni <aireni@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -152,28 +154,16 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
...
 # Revert the changes
 ifeq ($(TARGET), k8s)
-	@cd config/network && $(KUSTOMIZE) edit remove resource ingress.yaml
-	@sed -i 's/$(HOST)/REPLACE_HOSTNAME/' ./config/network/ingress.yaml
-else ifeq ($(TARGET), ocp)
-	@cd config/network && $(KUSTOMIZE) edit remove resource route.yaml
-	@cd config/prometheus && $(KUSTOMIZE) edit remove resource prometheus_role.yaml
-	@cd config/prometheus && $(KUSTOMIZE) edit remove resource prometheus_role_binding.yaml
-	@cd config/prometheus && $(KUSTOMIZE) edit remove resource monitor.yaml
+	@sed -i 's/$(HOST)/REPLACE_HOSTNAME/' ./config/k8s/network/ingress.yaml
```
Comment (masayag) [inline Makefile:166]: Is there an option to avoid using sed to manipulate the file?

File: Makefile [hunk 2]
```diff
@@ -140,6 +140,8 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 	$(KUSTOMIZE) build config/crd | kubectl delete -f -
 
 deploy: gen-manifests ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.7.1/cert-manager.yaml
```
Comment (masayag) [inline Makefile:143]: pls make sure we revert the changes in `make undeploy`

Comment (arielireni) [reply Makefile:143]: done

File: Makefile [hunk 3]
```diff
@@ -148,32 +150,25 @@ endif
 endif
 
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/default | kubectl delete -f -
+	kubectl delete -f https://github.com/cert-manager/cert-manager/releases/download/v1.7.1/cert-manager.yaml
```
[the remark this answers is not in the document]
Comment (arielireni) [reply Makefile]: done

File: config/k8s/network/ingress.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   name: ingress
-  namespace: system
+  namespace: flotta
```
Comment (masayag) [inline config/k8s/network/ingress.yaml:5]: I can only assume that the same will have to be changed here as well:

https://github.com/project-flotta/flotta-operator/blob/main/config/network/route.yaml#L7

Comment (arielireni) [reply config/k8s/network/ingress.yaml:5]: done

File: config/ocp/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+bases:
+  - ../../default
```
Comment (masayag) [inline config/ocp/kustomization.yaml:2]: there is an extra `../` in the path. also, bases is deprecated. move directly to be the first value of the `resources`.

**Review by masayag — CHANGES_REQUESTED** [review]

**Review by masayag — APPROVED** [review]

**Review by pkliczewski — APPROVED** [review]
